### PR TITLE
Updater stuck at 'Finalizing updates' during manual testing #461

### LINKED ISSUE #461: Updater stuck at 'Finalizing updates' during manual testing
During manual testing, the footer status bar stays stuck at 'Finalizing updates'.
OS: Windows
image

### PR DESCRIPTION
**Pull request by yl-coder**
Fixes #461

### CODE REVIEW

**Comment by damithc** [comment]
@HansNewbie has a few PRs open, so I'm not sure if this will conflict with them. May be he fixed this already.

**Comment by yl-coder** [comment]
Ok, let wait till he merged his PR and review this PR.

**Comment by damithc** [comment]
@HansNewbie can review this PR?

**Comment by HansNewbie** [comment]
Will do

File: src/main/java/address/updater/UpdateManager.java [hunk 1]
```diff
@@ -164,6 +171,10 @@ private void checkForUpdate() {
         downloadedVersions.add(latestVersion.get());
     }
 
+    private boolean isJarUpdaterExist() {
```
Comment (HansNewbie) [inline src/main/java/address/updater/UpdateManager.java]: Should be `isJarUpdaterResourceExist` not to confuse with the extracted file. And should return positive value if it exists (negate at the `if` caller).

File: src/main/java/address/updater/UpdateManager.java [hunk 2]
```diff
@@ -149,7 +150,13 @@ private void checkForUpdate() {
             logger.debug(MSG_FAIL_CREATE_UPDATE_SPEC);
             return;
         }
-
+        
+        if (isJarUpdaterExist()) {
+            raise(new UpdaterFailedEvent(MSG_JAR_UPDATER_MISSING));
+            logger.debug(MSG_JAR_UPDATER_MISSING);
```
Comment (HansNewbie) [inline src/main/java/address/updater/UpdateManager.java]: This should be `logger.fatal()`. `JarUpdater` resource should always exist.

Comment (yl-coder) [inline src/main/java/address/updater/UpdateManager.java]: noted

File: src/main/java/address/updater/UpdateManager.java (hunk 1 above)
Comment (yl-coder) [inline src/main/java/address/updater/UpdateManager.java]: noted

**Pull request merged**